ethpm_types/utils.py

```diff
@@ -89,6 +89,47 @@ def _sort(value: Any) -> Any:
     return json.dumps(sorted_settings, separators=(",", ":"), sort_keys=True)
 
 
+def parse_signature(sig: str) -> Tuple[str, List[Tuple[str, str, str]], List[str]]:
+    """
+    Parse an event or function signature into name and inputs
+
+    Args:
+        sig (str): The string signature to parse.
+
+    Returns:
+        A tuple of (name, inputs, outputs) where inputs is a list of tuples of (type, indexed, arg
+        name) and ouputs is a list of types.
+    """
+    outsplit = sig.split(" -> ")
+    std_sig = outsplit[0]
+    outputs_maybe = ""
+    if len(outsplit) > 1:
+        outputs_maybe = outsplit[1]
+    name, remainder = std_sig.split("(")
+    input_tups = [
+        tuple(y.strip().split(" "))
+        for y in filter(lambda x: x, [x for x in remainder.rstrip(")").split(",")])
+    ]
+    inputs = []
+    outputs = []
+
+    for intup in input_tups:
+        inlen = len(intup)
+        if inlen == 2:
+            inputs.append((intup[0], "", intup[1]))
+        elif inlen == 3 and intup[1] == "indexed":
+            assert len(intup) == 3  # mypy more like mywhy
```

Comment:
idea: could maybe avoid if you annotate `input_tups`

Reply:
It's a list of tuples and thruples.  Would an annotation help here?  Already doing a length check on this block but mypy doesn't pick up on it.

Reply:
It is what I would do to try and fix mypy in this situation but I don't know if it actually helps or is better.

Reply:
Wasn't able to immediately find a good annotation solution, let alone run it by mypy.  Since `spli()` returns a tuple of 1 to infinity.  If you got a specific idea, lmk.  Think the only good typed solution is hide it all behind another parsing function that has a well-defined return type to act as a kind of type guard.